ingester/ingester.go

```diff
@@ -165,6 +174,7 @@ func New(cfg Config, chunkStore cortex_chunk.Store) (*Ingester, error) {
 			Help: "The total number of samples returned from queries.",
 		}),
 	}
+	prometheus.MustRegister(i)
```

Comment:
@tomwilkie Registering Prometheus metrics from a constructor is problematic because it makes it impossible to instantiate multiple objects of the same type (the Prometheus client library will panic). I'm hitting this with ingester tests right now. While there's a non-panicking `Register` method, neither returning its error nor letting it silently fail seem like the right choice.

I would propose either moving this out of the constructor again or making the metrics global. Which one do you prefer?